griptape/tasks/prompt_task.py

```diff
@@ -92,54 +92,30 @@ def prompt_stack(self) -> PromptStack:
         stack = PromptStack(tools=self.tools)
         memory = self.structure.conversation_memory if self.structure is not None else None
 
-        system_template = self.generate_system_template(self)
-        if system_template:
-            stack.add_system_message(system_template)
+        rulesets = self.rulesets
+        system_artifacts = [TextArtifact(self.generate_system_template(self))]
+        if self.prompt_driver.use_native_structured_output:
+            self._add_native_schema_to_prompt_stack(stack, rulesets)
+
+        # Ensure there is at least one Ruleset that has non-empty `rules`.
+        if any(len(ruleset.rules) for ruleset in rulesets):
+            system_artifacts.append(TextArtifact(J2("rulesets/rulesets.j2").render(rulesets=rulesets)))
+
+        # Ensure there is at least one system Artifact that has a non-empty value.
+        has_system_artifacts = any(system_artifact.value for system_artifact in system_artifacts)
+        if has_system_artifacts:
+            stack.add_system_message(ListArtifact(system_artifacts))
```

Comment:
By separating out the rendering of Rules from `generate_system_template` it solves #1467 but only in the context of Prompt Task. There are plenty of other places where we use Rules ([RagEngine](https://github.com/griptape-ai/griptape/blob/d07d699bc22e23ff464036972d3a5cfcbb622316/griptape/engines/rag/modules/response/prompt_response_rag_module.py#L59-L60)) that would not benefit from this change.

Open to discussion on how we can solve this globally.